### PR TITLE
refactor: change alignment implementation via compiler plugin to be debug-friendly (issue #337)

### LINKED ISSUE #337: Debug-friendly code generation
The plugin currently generates stuff like:

```kotlin
alignedOn("function name") {
    actualCode
}
```

when `actualCode` contains aggregate code. I propose to generate instead:
```kotlin
alignRaw("function name")
val mangled = actualCode
dealign()
return mangled
```

Otherwise, debugging can be hell:
image

### PR DESCRIPTION
**Pull request by nicolasfara**
This PR changes the way the alignment is performed via the compiler plugin.
From now on, the plugin calls the `alignRaw` function before executing an aggregate block, and after the block, `dealing` is called.

Closes #337

### CODE REVIEW

File: dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt [hunk 1]
```diff
@@ -89,4 +89,14 @@ interface Aggregate<ID : Any> {
      * Returns the body's return element.
      */
     fun <R> alignedOn(pivot: Any?, body: () -> R): R
+
+    /**
+     * Pushes the pivot in the alignment stack.
+     */
+    fun alignRaw(pivot: Any?)
```
Comment (DanySK) [inline dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt:96]: Is it possible to hide the function but still call it with plugin magic?

**Review by DanySK — APPROVED** [review]

File: dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt (hunk 1 above)
Comment (nicolasfara) [reply dsl/src/commonMain/kotlin/it/unibo/collektive/aggregate/api/Aggregate.kt:96]: I've tried, but no way...

**Review by angelacorte — APPROVED** [review]

File: compiler-embeddable/src/test/kotlin/it/unibo/collektive/test/TestCompileString.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ class TestCompileString : FreeSpec({
                 disassembled shouldNot beNull()
                 disassembled shouldNot beBlank()
                 val alignedOnCalls = disassembled.lines().filter {
-                    "// InterfaceMethod it/unibo/collektive/aggregate/api/Aggregate.alignedOn:" in it
+                    "// InterfaceMethod it/unibo/collektive/aggregate/api/Aggregate.alignRaw:" in it
```
Comment (DanySK) [inline compiler-embeddable/src/test/kotlin/it/unibo/collektive/test/TestCompileString.kt:37]:
```suggestion
                    "// InterfaceMethod it/unibo/collektive/aggregate/api/Aggregate.align" in it
```

This should work also if we decide to switch back

**Review by DanySK — APPROVED** [review]
See if it work with my change, and then just squash this in.

**Comment by DanySK** [comment]
Also at this point "alignRaw" should become simply "align".

**Pull request merged**